include/bm/bm_sim/calculations.h

```diff
@@ -387,6 +387,32 @@ class CustomCrcMgr {
                                           const crc_config_t &config);
 };
 
+enum class RssErrorCode {
+  SUCCESS = 0,
+  INVALID_CALCULATION_NAME,
+  WRONG_TYPE_CALCULATION,
+  INVALID_KEY,
+};
+
+namespace detail {
+
+using rss_key_t = std::vector<uint8_t>;
```

Comment:
could you remind me why this is not `std::vector<uint32_t>`, given that you require `len % 4 != 0`?

Reply:
My main concern is the ambiguity of endianness. I think it puts less mental burden on users this way. Otherwise, users might need to worry about the endianness of `uint32_t` when trying to set the RSS key.

Reply:
sounds good